src/ext/coms.c

```diff
@@ -633,12 +634,23 @@ static void _dd_curl_set_connect_timeout(CURL *curl) {
 }
 
 static void _dd_curl_set_hostname(CURL *curl) {
+    char *url = get_dd_trace_agent_url();
+    if (url && url[0]) {
+        size_t agent_url_len = strlen(url) + sizeof(TRACE_PATH_STR);
+        char *agent_url = malloc(agent_url_len);
+        sprintf(agent_url, "%s%s", url, TRACE_PATH_STR);
+        curl_easy_setopt(curl, CURLOPT_URL, agent_url);
+        free(url);
+        free(agent_url);
+        return;
+    }
+
```

Comment:
Just woke up so I'm still a bit groggy, but I think the memory leak might be due to a zero-length string not getting freed. A zero-length string won't pass the `if` check, but it will still need freed... I think.